core/src/main/scala/org/apache/spark/ui/JettyUtils.scala

```diff
@@ -308,14 +308,22 @@ private[spark] object JettyUtils extends Logging {
 
       sslOptions.createJettySslContextFactory().foreach { factory =>
         // If the new port wraps around, do not try a privileged port.
+
+        require(sslOptions.port == 0 || (1024 <= sslOptions.port && sslOptions.port < 65536))
+
         val securePort =
           if (currentPort != 0) {
-            (currentPort + 400 - 1024) % (65536 - 1024) + 1024
+            if (1024 < sslOptions.port && sslOptions.port < 65536) {
```

Comment:
This condition should only happen when the port is 0 right? given the `require` above. It would be clearer to check against 0 only.